src/libs/collect.c

```diff
@@ -1838,6 +1840,18 @@ static void list_view(dt_lib_collect_rule_t *dr)
         }
         break;
 
+      case DT_COLLECTION_PROP_RATING: // image rating
+        {
+          g_snprintf(query, sizeof(query),
+                     "SELECT (flags & 7) AS rating,"
```

Comment:
to handle rejected images (and so be able to ignore them) you can do:`

SELECT CASE WHEN (flags & 8) == 8 THEN -1 ELSE (flags & 7) END AS ratings,`

Reply:
Just a side note that this is not tested, just a hint!

Reply:
Thanks for the hint, just a minor typo but otherwise works :)
The full filtering solution is a bit more involved, see my answer below.